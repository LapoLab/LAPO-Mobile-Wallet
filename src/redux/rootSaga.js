import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
