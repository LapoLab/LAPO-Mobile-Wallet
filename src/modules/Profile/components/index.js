/**
 * @flow
 */

import React, { Component } from 'react';
import {
  Text,
  View,
  Image,
  TouchableOpacity,
  SafeAreaView,
  TextInput,
  Keyboard,
  Platform,
  BackHandler,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import LinearGradient from 'react-native-linear-gradient';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import ImagePicker from 'react-native-image-picker';

import styles from './styles';
import Button from '../../../commons/components/Button';
import * as actions from '../../../commons/actions';
import * as actionsAlert from '../../Alert/actions';
import { backIcon, fotoAparat, iconEdit } from '../../../assets';

type Props = {
  navigation: { openDrawer: Function, dispatch: Function },
  avatarSource: { uri: string },
  // eslint-disable-next-line
  password: string,
  // eslint-disable-next-line
  username: string,
  isOpenDrawer: boolean,
  setAvatar: Function,
  setUserName: Function,
  setPins: Function,
  setConfirmPins: Function,
  goToOtherWindow: Function,
  showAlert: Function,
};
type State = {
  keyboardShow: boolean,
  avatarSource: { uri: string },
  password: string,
  username: string,
  isEdit: boolean,
};
class Profile extends Component<Props, State> {
  state = {
    keyboardShow: false,
    avatarSource: { uri: '' },
    password: '',
    username: '',
    // eslint-disable-next-line
    isEdit: false,
  };

  static getDerivedStateFromProps(props: Props, state: State) {
    if (!state.isEdit) {
      return {
        ...state,
        avatarSource: props.avatarSource,
        password: props.password,
        username: props.username,
      };
    }
    return state;
  }

  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBack);
    this.keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', this._keyboardDidShow);
    this.keyboardDidShowListener = Keyboard.addListener('keyboardWillShow', this._keyboardDidShow);
    this.keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', this._keyboardDidHide);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBack);
    this.keyboardDidShowListener.remove();
    this.keyboardDidHideListener.remove();
  }

  handleBack = async () => {
    const { isOpenDrawer } = this.props;
    if (isOpenDrawer) {
      this.navigateToScreen('Dashboard')();
      return true;
    }
    this.handleGoToBack();
    return true;
  };

  // eslint-disable-next-line react/sort-comp
  keyboardDidShowListener: Function;

  keyboardDidHideListener: Function;

  _keyboardDidShow = () => {
    this.setState({ keyboardShow: true });
  };

  _keyboardDidHide = () => {
    this.setState({ keyboardShow: false });
  };

  handleGoToBack = () => {
    const {
      navigation: { openDrawer },
    } = this.props;
    openDrawer();
  };

  navigateToScreen = route => () => {
    const {
      navigation: { dispatch },
    } = this.props;
    const navigateAction = NavigationActions.navigate({
      routeName: route,
    });
    dispatch(navigateAction);
  };

  textInputPassword: Object;

  textInputName: Object;

  handleSaveProfile = () => {
    const {
      setAvatar, setUserName, setPins, setConfirmPins, showAlert,
    } = this.props;
    const { username, password, avatarSource } = this.state;

    if (password.length !== 6) {
      showAlert(
        'Message',
        'passcode not valid',
        // eslint-disable-next-line
        [{ text: 'OK' }],
        {
          cancelable: false,
        },
      );
      return;
    }

    setPins(password);
    setConfirmPins(password);

    setUserName(username);
    setAvatar(avatarSource);

    // eslint-disable-next-line
    this.setState({ isEdit: false });

    this.navigateToScreen('Dashboard')();
  };

  handleEditPassword = () => {
    this.textInputPassword.focus();
  };

  handleEditName = () => {
    this.textInputName.focus();
  };

  handleChangePassword = (value: string) => {
    // eslint-disable-next-line
    this.setState({ password: value, isEdit: true });
  };

  handleChangeUserName = (value: string) => {
    // eslint-disable-next-line
    this.setState({ username: value, isEdit: true });
  };

  handleShowImagePicker = async () => {
    const { goToOtherWindow } = this.props;
    // eslint-disable-next-line
    await this.setState({ isEdit: true });

    goToOtherWindow(true);

    const options = {
      title: 'Select Avatar',
      storageOptions: {
        skipBackup: true,
        path: 'images',
      },
    };

    ImagePicker.showImagePicker(options, (response) => {
      goToOtherWindow(true);
      // console.log('response:::::', response);
      if (response.didCancel) {
        // console.log('User cancelled image picker');
      } else if (response.error) {
        // console.log('ImagePicker Error: ', response.error);
      } else {
        const source = { uri: Platform.OS === 'ios' ? response.uri : `file://${response.path}` };
        this.setState({
          avatarSource: source,
        });
      }
    });
  };

  render() {
    const {
      keyboardShow, avatarSource, username,
    } = this.state;

    return (
      <LinearGradient
        start={{ x: -0.1, y: -0.1 }}
        end={{ x: 1.2, y: 1.2 }}
        locations={[0.04, 0.96]}
        colors={['#4183BD', '#093B0C']}
        style={{ flex: 1 }}
      >
        <SafeAreaView style={styles.container}>
          <View style={styles.header}>
            <TouchableOpacity onPress={this.handleGoToBack}>
              <Image style={styles.backHeader} source={backIcon} />
            </TouchableOpacity>
            <Text style={styles.titleHeader}>Your profile</Text>
          </View>

          {!keyboardShow && (
            <View style={styles.fotoContainer}>
              <View style={styles.fotoProxy}>
                <View style={avatarSource && avatarSource.uri ? '' : styles.imgContainerWithBorder}>
                  <Image style={styles.imgFota} source={avatarSource} />
                </View>
                <TouchableOpacity
                  style={styles.fotoAparatContainer}
                  onPress={this.handleShowImagePicker}
                >
                  <Image style={styles.fotoAparat} source={fotoAparat} />
                </TouchableOpacity>
              </View>
            </View>
          )}

          <View style={styles.inputContainer}>
            <Text style={styles.label}>Name</Text>
            <View style={styles.rowInput}>
              <TextInput
                style={styles.input}
                value={username}
                autoCapitalize="words"
                textContentType="username"
                // $FlowFixMe
                ref={ref => (this.textInputName = ref)}
                onChangeText={this.handleChangeUserName}
              />
              <TouchableOpacity style={styles.buttonEdit} onPress={this.handleEditName}>
                <Image style={styles.iconEdit} source={iconEdit} />
              </TouchableOpacity>
            </View>
          </View>

          <View style={styles.line} />

          <View style={styles.inputContainer}>
            <Text style={styles.label}>Your passcode</Text>
            <View style={styles.rowInput}>
              <TextInput
                style={styles.input}
                // defaultValue={password}
                secureTextEntry
                textContentType="password"
                keyboardType="number-pad"
                returnKeyType="done"
                maxLength={6}
                autoCorrect={false}
                // $FlowFixMe
                ref={ref => (this.textInputPassword = ref)}
                onChangeText={this.handleChangePassword}
              />
              <TouchableOpacity style={styles.buttonEdit} onPress={this.handleEditPassword}>
                <Image style={styles.iconEdit} source={iconEdit} />
              </TouchableOpacity>
            </View>
          </View>

          <View style={styles.line} />

          {!keyboardShow && (
            <Text style={styles.info}>
              Please note: all this information are stored ONLY in your cellphone. We are not
              storing any of your personal data and/or picture.
            </Text>
          )}

          <View style={{ flex: 2 }}>
            <Button title="SAVE PROFILE" onPress={this.handleSaveProfile} status="activeLight" />
          </View>
        </SafeAreaView>
      </LinearGradient>
    );
  }
}

const mapStateToProps = state => ({
  avatarSource: state.settings.avatar,
  username: state.settings.username,
  password: state.pins.pins,
  isOpenDrawer: state.modals.isOpenDrawer,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    setUserName: actions.setUserName,
    setAvatar: actions.setAvatar,
    setPins: actions.setPins,
    setConfirmPins: actions.setConfirmPins,
    goToOtherWindow: actions.goToOtherWindow,
    showAlert: actionsAlert.showAlert,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Profile);
