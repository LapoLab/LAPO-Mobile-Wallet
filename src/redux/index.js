import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
