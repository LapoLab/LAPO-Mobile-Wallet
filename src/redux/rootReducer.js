import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
