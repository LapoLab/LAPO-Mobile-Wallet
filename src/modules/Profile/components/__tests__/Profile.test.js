// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import Profile from '../index';

describe('Profile page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
