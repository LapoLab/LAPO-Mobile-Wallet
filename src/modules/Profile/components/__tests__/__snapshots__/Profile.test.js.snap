// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page renders snapshots 1`] = `
<BVLinearGradient
  colors={
    Array [
      -12483651,
      -16172276,
    ]
  }
  endPoint={
    Object {
      "x": 1.2,
      "y": 1.2,
    }
  }
  locations={
    Array [
      0.04,
      0.96,
    ]
  }
  startPoint={
    Object {
      "x": -0.1,
      "y": -0.1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "flex-start",
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/back.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "marginLeft": 12,
              "resizeMode": "contain",
              "width": 12,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "white",
            "flex": 1,
            "fontFamily": "Lato-Bold",
            "fontSize": 18,
            "marginRight": 24,
            "textAlign": "center",
          }
        }
      >
        Your profile
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "borderColor": "white",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 100,
              "width": 100,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "borderRadius": 50,
                "height": 100,
                "resizeMode": "cover",
                "width": 100,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "left": 80,
              "opacity": 1,
              "position": "absolute",
              "top": "65%",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/fotoAparat.png",
              }
            }
            style={
              Object {
                "resizeMode": "contain",
                "width": 25,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "height": 60,
          "justifyContent": "center",
          "paddingHorizontal": 35,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Lato-Bold",
            "fontSize": 20,
            "marginVertical": 10,
          }
        }
      >
        Name
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 30,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="words"
          onChangeText={[Function]}
          style={
            Object {
              "color": "white",
              "flex": 8,
              "fontFamily": "Lato-Regular",
              "fontSize": 16,
              "minHeight": 30,
              "padding": 0,
            }
          }
          textContentType="username"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
              "width": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/iconEdit.png",
              }
            }
            style={
              Object {
                "resizeMode": "contain",
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderTopColor": "#C9C9C9",
          "borderTopWidth": 1,
          "marginVertical": 20,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "height": 60,
          "justifyContent": "center",
          "paddingHorizontal": 35,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Lato-Bold",
            "fontSize": 20,
            "marginVertical": 10,
          }
        }
      >
        Your passcode
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 30,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          keyboardType="number-pad"
          maxLength={6}
          onChangeText={[Function]}
          returnKeyType="done"
          secureTextEntry={true}
          style={
            Object {
              "color": "white",
              "flex": 8,
              "fontFamily": "Lato-Regular",
              "fontSize": 16,
              "minHeight": 30,
              "padding": 0,
            }
          }
          textContentType="password"
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
              "width": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/iconEdit.png",
              }
            }
            style={
              Object {
                "resizeMode": "contain",
                "width": 20,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderTopColor": "#C9C9C9",
          "borderTopWidth": 1,
          "marginVertical": 20,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "flex": 2,
          "fontFamily": "Lato-Regular",
          "fontSize": 18,
          "paddingHorizontal": 40,
        }
      }
    >
      Please note: all this information are stored ONLY in your cellphone. We are not storing any of your personal data and/or picture.
    </Text>
    <View
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#4183BD",
              "borderColor": "rgb(42, 101, 117)",
              "borderRadius": 74,
              "borderWidth": 8,
              "height": 64,
              "justifyContent": "center",
              "marginHorizontal": 25,
              "opacity": 1,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Lato-Regular",
                "fontSize": 18,
              }
            }
          >
            SAVE PROFILE
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
